lib/matplotlib/tests/test_path.py

```diff
@@ -88,25 +88,27 @@ def test_contains_points_negative_radius():
     np.testing.assert_equal(result, [True, False, False])
 
 
-_test_paths = [
+_ExampleCurve = namedtuple('_ExampleCurve', ['path', 'extents', 'area'])
+_test_curves = [
     # interior extrema determine extents and degenerate derivative
-    Path([[0, 0], [1, 0], [1, 1], [0, 1]],
-           [Path.MOVETO, Path.CURVE4, Path.CURVE4, Path.CURVE4]),
-    # a quadratic curve
-    Path([[0, 0], [0, 1], [1, 0]], [Path.MOVETO, Path.CURVE3, Path.CURVE3]),
+    _ExampleCurve(Path([[0, 0], [1, 0], [1, 1], [0, 1]],
+                       [Path.MOVETO, Path.CURVE4, Path.CURVE4, Path.CURVE4]),
+                  extents=(0., 0., 0.75, 1.), area=0.6),
+    # a quadratic curve, clockwise
+    _ExampleCurve(Path([[0, 0], [0, 1], [1, 0]],
+                       [Path.MOVETO, Path.CURVE3, Path.CURVE3]),
+                  extents=(0., 0., 1., 0.5), area=-1/3),
     # a linear curve, degenerate vertically
-    Path([[0, 1], [1, 1]], [Path.MOVETO, Path.LINETO]),
+    _ExampleCurve(Path([[0, 1], [1, 1]], [Path.MOVETO, Path.LINETO]),
+                  extents=(0., 1., 1., 1.), area=0.),
     # a point
-    Path([[1, 2]], [Path.MOVETO]),
+    _ExampleCurve(Path([[1, 2]], [Path.MOVETO]), extents=(1., 2., 1., 2.),
+                  area=0.),
 ]
```

Comment:
Test a simple non-curved polygon such as a triangle:
```suggestion
    _ExampleCurve(Path([[1, 1], [2, 1], [1.5, 2]]), extents=(1, 1, 2, 2), area=1/3),
]
```
Side-remark: Not an expert, but I believe there are more efficient formulas for simple polygons without Bezier segments.

Reply:
I agree, but looking through this it is all vectorized and over the degree-1 shapes I'm guessing it isn't _that_ expensive. I think this should also be a good way for someone to later come in and add a fast-path for those cases: `if CURVE == ...: simple formula`. Starting with this seems reasonable to me.

Reply:
Digression, but this is apparently messing up mypy stubtest which I think is due to the float in here vs int in all the other paths. Changing this to `tuple(point)` instead satisfies mypy. Seems like typing gone wrong here.

Do we really need to be running mypy over the test suite?

Reply:
I don't have a strong opinion on mypy on tests. I assume the added value would be that we check for reasonable typing of our interfaces by exercising calls to these interfaces.